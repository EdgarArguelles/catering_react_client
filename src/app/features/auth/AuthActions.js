/**
 * Following the Flux Standard Action: https://github.com/acdlite/flux-standard-action
 */
import keyMirror from 'keymirror';
import Api from '../../common/Api';

export const ACTION_TYPES = keyMirror({
  SOCKET_CONNECTED: null,
  LOGOUT: null,
  LOGIN_SUCCESS: null,
  PING_USER_SUCCESS: null,
  PING_USER_ERROR: null,
});

/**
 * Remove "accessToken" and "userImage" from sessionStorage
 *
 * @return {void}
 */
const removeToken = () => {
  window.sessionStorage.removeItem('accessToken');
  window.sessionStorage.removeItem('userImage');
};

/**
 * Store "accessToken" and "userImage" in sessionStorage
 *
 * @param {Object} loggedUser user information
 * @return {void}
 */
const saveToken = loggedUser => {
  window.sessionStorage.setItem('accessToken', loggedUser.token);
  loggedUser.image && window.sessionStorage.setItem('userImage', loggedUser.image);
};

export default class AuthActions {
  static connectSocket() {
    return {
      type: ACTION_TYPES.SOCKET_CONNECTED,
    };
  }

  static logout() {
    removeToken();
    return {
      type: ACTION_TYPES.LOGOUT,
    };
  }

  static login(body) {
    saveToken(body.loggedUser);
    return {
      type: ACTION_TYPES.LOGIN_SUCCESS,
      payload: {
        loggedUser: body.loggedUser,
      },
    };
  }

  static fetchPing() {
    const body = {query: '{ping {id fullName image role token permissions}}'};

    return async dispatch => {
      try {
        const json = await Api.graphql(dispatch, body);
        const loggedUser = json && json.data && json.data.ping && json.data.ping.id ? json.data.ping : null;
        if (loggedUser) {
          saveToken(loggedUser);
          dispatch({
            type: ACTION_TYPES.PING_USER_SUCCESS,
            payload: {
              loggedUser,
            },
          });
        } else {
          removeToken();
          dispatch({type: ACTION_TYPES.PING_USER_ERROR});
        }
        return json;
      } catch (error) {
        removeToken();
        dispatch({type: ACTION_TYPES.PING_USER_ERROR});
        throw error;
      }
    };
  }
}