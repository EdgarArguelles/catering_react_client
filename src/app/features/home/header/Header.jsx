import './Header.scss';
import React, {useEffect, useState} from 'react';
import Grid from '@material-ui/core/Grid';
import Logo from './logo/Logo';
import Menu from './menu/Menu';

const Header = () => {
  const [fixed, setFixed] = useState(false);
  useEffect(() => {
    const handleScroll = () => setFixed(window.pageYOffset > 10);
    window.addEventListener('scroll', handleScroll);

    return () => window.removeEventListener('scroll', handleScroll);
  }, []);

  return (
    <header id="header" className={fixed ? 'header-fixed' : ''}>
      <Grid container className="header-container" justify="space-between">
        <Logo id="header-logo"/>
        <Menu/>
      </Grid>
    </header>
  );
};

export default React.memo(Header);