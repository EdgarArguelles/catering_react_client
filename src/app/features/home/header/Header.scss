@use 'app/Variables';

#header {
  padding: 50px 0;
  height: 92px;
  position: fixed;
  left: 0;
  top: 0;
  right: 0;
  transition: all 0.5s;
  z-index: 1;

  &.header-fixed {
    background: Variables.$header-color;
    padding: 30px 0;
    height: Variables.$header-fixed-height;

    .header-container {
      margin-top: -15px;

      #header-logo {
        cursor: pointer;
        width: 80px;
        height: 80px;
        top: -10px;
        left: 35px;
        transform: translate(0, 0);

        @media (max-width: Variables.$screen-xs) {
          left: 10px;
        }
      }
    }
  }
}