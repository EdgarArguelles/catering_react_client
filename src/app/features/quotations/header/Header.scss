@use 'app/Variables' as *;

$header-height: 64px;
$screen-menu: 734px;

#quotations-header {
  height: $header-height;

  #app-bar {
    height: $header-height;
    background: $header-color;
    box-shadow: none;

    .tool-bar {
      padding: 0 24px;

      @media (max-width: $screen-menu) {
        padding: 0 10px;
      }

      svg {
        margin-right: 5px;
      }
    }
  }
}