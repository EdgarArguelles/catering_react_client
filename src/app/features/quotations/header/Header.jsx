import './Header.scss';
import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Navigation from './navigation/Navigation';
import Menu from './menu/Menu';

const Header = () => {
  return (
    <header id="quotations-header">
      <AppBar id="app-bar">
        <Toolbar className="tool-bar">
          <Navigation/>
          <Menu/>
        </Toolbar>
      </AppBar>
    </header>
  );
};

export default React.memo(Header);