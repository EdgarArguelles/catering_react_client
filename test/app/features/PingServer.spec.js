import {expect} from 'chai';
import sinon from 'sinon';
import renderer from 'react-test-renderer';
import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import pingServer from 'app/features/PingServer';
import AuthActions from 'app/features/auth/AuthActions';
import Header from 'app/features/home/header/Header.react';

describe('PingServer', () => {
  const PingHeader = pingServer(Header);
  const dispatchStub = sinon.stub();
  const fetchPingStub = sinon.stub(AuthActions, 'fetchPing');
  let component, store, wrapper;

  afterEach(() => {
    dispatchStub.reset();
    fetchPingStub.reset();
  });

  const getProvider = children => {
    return <Provider store={store}>{children}</Provider>;
  };

  const mountComponent = () => {
    store = configureStore()({});
    store.dispatch = dispatchStub;
    wrapper = renderer.create(getProvider(<PingHeader/>));
    component = wrapper.root.find(el => el.type.name === 'PingServer');
  };

  describe('load', () => {
    it('should have all props', () => {
      mountComponent();

      expect(component.props.ping).to.not.be.undefined;
    });
  });

  describe('componentWillMount', () => {
    it('should call fetchPing', () => {
      fetchPingStub.withArgs().returns('response');
      mountComponent();

      sinon.assert.callCount(fetchPingStub, 1);
      sinon.assert.calledWithExactly(fetchPingStub);
      sinon.assert.callCount(dispatchStub, 1);
      sinon.assert.calledWithExactly(dispatchStub, 'response');
    });
  });

  describe('componentWillUpdate', () => {
    it('should call fetchPing', () => {
      fetchPingStub.withArgs().returns('response');
      mountComponent();
      wrapper.update(getProvider(<PingHeader test={true}/>));

      sinon.assert.callCount(fetchPingStub, 2);
      sinon.assert.calledWithExactly(fetchPingStub);
      sinon.assert.calledWithExactly(fetchPingStub);
      sinon.assert.callCount(dispatchStub, 2);
      sinon.assert.calledWithExactly(dispatchStub, 'response');
      sinon.assert.calledWithExactly(dispatchStub, 'response');
    });
  });

  describe('snapshot', () => {
    it('should render header', () => {
      mountComponent();

      global.expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });
});