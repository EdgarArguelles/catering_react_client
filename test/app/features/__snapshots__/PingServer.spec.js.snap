// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PingServer snapshot should render header 1`] = `
<header
  id="header"
>
  <div
    className="MuiGrid-root header-container MuiGrid-container MuiGrid-justify-xs-space-between"
  >
    <div
      id="logo"
    >
      <a
        href="#home"
      >
        <img
          className="company-logo"
          id="logo"
          src={
            Object {
              "\\">": "\\">",
              "145661": "145661",
              "3-c011": "3-c011",
              "org": "org",
              "w3": "w3",
            }
          }
        />
      </a>
    </div>
    <nav
      id="nav-menu-container"
    >
      <ul
        className="nav-menu"
      >
        <li
          className="menu-active"
        >
          <a
            href="#home"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="#about"
          >
            Quienes Somos
          </a>
        </li>
        <li>
          <a
            href="#services"
          >
            Servicios
          </a>
        </li>
        <li>
          <a
            href="#quotations"
          >
            Presupuesto
          </a>
        </li>
        <li>
          <a
            href="#contact"
          >
            Contactanos
          </a>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;
